core/src/display_object.rs

```diff
@@ -569,6 +641,19 @@ pub trait TDisplayObject<'gc>:
     /// it to the bounding box. This gives a tighter AABB then if we simply transformed
     /// the overall AABB.
     fn bounds_with_transform(&self, matrix: &Matrix) -> BoundingBox {
+        // A scroll rect completely overrides an object's bounds,
+        // and can even the bounding box to be larger than the actual content
+        if let Some(scroll_rect) = self.scroll_rect() {
+            return BoundingBox {
+                x_min: -scroll_rect.x_min,
+                y_min: scroll_rect.y_min,
+                x_max: scroll_rect.x_max,
+                y_max: scroll_rect.y_max,
+                valid: true,
```

Comment:
Hmm, actually this seems incorrect -- the bounds that get transformed in this method should be the non-transformed internal bounds of the object, i.e. the same as `this.getBounds(this)`. When `scrollRect` is on, the bounds start at (0, 0), so I believe this should be:
```
x_min: 0,
y_min: 0,
x_max: scroll_rect.x_max - scroll_rect.x_min,
y_max: scroll_rect.y_max - scroll_rect.y_min,
```

Reply:
Flash player actually does this for `getBounds` - after setting the `scrollRect` in the test swf, we get:
`circle.getBounds(stage) = (x=-33, y=-44, w=550, h=660)`

Reply:
That is after being transformed by the local-to-stage matrix (from the `stage` as the first parameter), which is taken care of by the `.transform(matrix)` step. The base rect that gets passed to `.transform(matrix)` should be the untransformed bounds.

For example, if I were to call `DisplayObject::get_bounds(&Default::default())`, I would expect this function to return the local object-space bounds of the object, which should be (0, 0)x(width, height) when the `scrollRect` is enabled.

Reply:
You're right - I've fixed this.